internal/store/store.go

```diff
@@ -10,19 +14,68 @@ import (
 
 // Store provides access to a store backed by the given reader.
 type Store struct {
-	plugins []*model.Plugin
+	plugins []pluginVersions
 	logger  logrus.FieldLogger
 }
 
+// pluginVersions is a sorted list of all versions of a given plugin.
+type pluginVersions []*model.Plugin
+
 // New constructs a new instance of Store.
 func New(reader io.Reader, logger logrus.FieldLogger) (*Store, error) {
 	plugins, err := model.PluginsFromReader(reader)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to parse stream")
 	}
 
+	pluginVersions, err := sortPlugins(plugins)
```

Comment:
It's unfortunate to have `pluginVersions` override the type with the same name here.